src/Microsoft.AspNet.Hosting/HostingContext.cs

```diff
@@ -21,7 +20,7 @@ public class HostingContext
         public Action<IApplicationBuilder> ApplicationStartup { get; set; }
         public RequestDelegate ApplicationDelegate { get; set; }
 
-        public string ServerName { get; set; }
+        public string ServerFactoryAssembly { get; set; }
```

Comment:
That this is the name of an assembly is an implementation detail of the default IServerFactory. How about ServerFactoryLocation?